eureka/src/main/java/com/linecorp/armeria/server/eureka/EurekaUpdatingListener.java

```diff
@@ -269,23 +275,44 @@ private static String vipAddress(@Nullable String vipAddress, String hostName, P
         return vipAddress != null ? vipAddress : hostName + ':' + portWrapper.getPort();
     }
 
+    @Nullable
+    private static String pageUrl(String hostnameOrIpAddr, @Nullable String pageUrlPath,
+                                  @Nullable String pageUrl, PortWrapper portWrapper) {
+        if (!isNullOrEmpty(pageUrl)) {
+            return pageUrl;
+        }
+        if (pageUrlPath == null || !portWrapper.isEnabled()) { // allow an empty String.
+            return null;
+        }
+        return concatPath(baseUrl(hostnameOrIpAddr, portWrapper, SessionProtocol.HTTP), pageUrlPath);
+    }
+
+    private static String baseUrl(String hostnameOrIpAddr, PortWrapper portWrapper,
+                                  SessionProtocol sessionProtocol) {
+        return sessionProtocol.uriText() + "://" +
+               hostnameOrIpAddr(hostnameOrIpAddr) + ':' + portWrapper.getPort();
+    }
+
+    private static String concatPath(String baseURL, String oldHealthCheckUrlPath) {
```

Comment:
style nit; since the method isn't specific to healthcheck

```suggestion
    private static String concatPath(String baseURL, String urlPath) {
```

Reply:
That's correct. Thanks!